scripting/weaponmodels/entitydata.sp

```diff
@@ -118,72 +128,147 @@ void WeaponModels_EntityDataInit()
+	InitSendPropOffset(g_iOffset_ViewModelOwner, "CBaseViewModel", "m_hOwner", true);
+	InitSendPropOffset(g_iOffset_ViewModelWeapon, "CBaseViewModel", "m_hWeapon", true);
+	InitSendPropOffset(g_iOffset_ViewModelSequence, "CBaseViewModel", "m_nSequence", true);
+	InitSendPropOffset(g_iOffset_ViewModelPlaybackRate, "CBaseViewModel", "m_flPlaybackRate", true);
+	InitSendPropOffset(g_iOffset_ViewModelIndex, "CBaseViewModel", "m_nViewModelIndex", true);
 	
 	InitSendPropOffset(g_iOffset_ViewModelIgnoreOffsAcc, "CBaseViewModel", "m_bShouldIgnoreOffsetAndAccuracy", false);
 	InitSendPropOffset(g_iOffset_EconItemDefinitionIndex, "CEconEntity", "m_iItemDefinitionIndex", false);
 
 	int lightingOriginOffset;
-	InitSendPropOffset(lightingOriginOffset, "CBaseAnimating", "m_hLightingOrigin");
+	InitSendPropOffset(lightingOriginOffset, "CBaseAnimating", "m_hLightingOrigin", true);
 
 	// StudioHdr offset in gameconf is only relative to the offset of m_hLightingOrigin, in order to make the offset more resilient to game updates
 	g_iOffset_Animating_StudioHdr += lightingOriginOffset;
 }
 
-int GetPlayerViewModel(int client, int index)
+public int GetPlayerViewModel(int client, int index)
 {
+	#if defined DEBUG
+	PrintToServer("GetPlayerViewModel");
+	#endif
 	return GetEntDataEnt2(client, g_iOffset_PlayerViewModel + (index * 4));
 }
 
-void SetPlayerViewModel(int client, int index, int viewModel)
+public void SetPlayerViewModel(int client, int index, int viewModel)
 {
+	#if defined DEBUG
+	PrintToServer("SetPlayerViewModel");
+	#endif
 	SetEntDataEnt2(client, g_iOffset_PlayerViewModel + (index * 4), viewModel, true);
 }
 
-void SetEntityVisibility(int entity, bool show)
-{
-	int flags = GetEntData(entity, g_iOffset_EntityEffects);
+public void SetEntityVisibility(int entity, bool show){
+
+	#if defined DEBUG
+	PrintToServer("Seting EntityVisibility for entity: %d", entity);
+	#endif
 
+	int flags = GetEntData(entity, g_iOffset_EntityEffects);
 	SetEntData(entity, g_iOffset_EntityEffects, show ? flags & ~EF_NODRAW : flags | EF_NODRAW, _, true);
 }
 
+public void NextFrameSetVisibility(DataPack dataPackHandle){
+
+	ResetPack(dataPackHandle);
+ 	int framestoskip = dataPackHandle.ReadCell();
+ 	if ( framestoskip > 0 )		//recursive frame skip	
+	{
+		ResetPack(dataPackHandle);
+ 		dataPackHandle.WriteCell( framestoskip-1 );
+		
+ 		RequestFrame(NextFrameSetVisibility, dataPackHandle);
+ 		return;
+	}
+	int show = dataPackHandle.ReadCell();
+	int entity = dataPackHandle.ReadCell();
+	
+	SetEntityVisibility(entity, show ? true : false);
+}
+
+
+
+//bug in nmrih: old viewmodel still visible: double viewmodel
+//perhaps caused by unknown function overwriting flags after delay
+//temp fix: add frame skip delay to setting viewmodel1's nodraw flag
+//TODO: Find cause and less hacky solution!
+public void SetEntityVisibility_FrameDelay(int entity, bool show){
+	DataPack pack = new DataPack();
+
+	pack.WriteCell(60);					//frames to skip
+
+	if (show)	pack.WriteCell(1);		 
```

Comment:
this code style does not conform to any other code in the project. Eventhough you might have a different style preference which is fine, anyone can agree on keeping it consistant through a project.